packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -7,6 +7,51 @@ import { mergePrincipal } from './util';
  * Represents a statement in an IAM policy document.
  */
 export class PolicyStatement {
+  /**
+   * Creates a new PolicyStatement based on the object provided.
+   * This will accept an object created from the `.toJSON()` call
+   * @param obj the PolicyStatement in object form.
+   */
+  public static fromJson(obj: any) {
+    const ensureArrayOrUndefined = (field: any) => {
+      if (field === undefined) {
+        return undefined;
+      }
+      if (typeof (field) !== "string" && !Array.isArray(field)) {
+        throw new Error("Fields must be either a string or an array of strings");
+      }
+      if (Array.isArray(field) && !field.reduce((acc: boolean, f: any) => (acc && typeof (f) === "string"), true)) {
```

Comment:
Why use `reduce` instead of `field.find(x => typeof(x) !== 'string')`?

Reply:
habit...

Reply:
simplified...